x-pack/plugins/enterprise_search/server/lib/get_oauth_token_package_params.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ENTERPRISE_SEARCH_KIBANA_COOKIE } from '../../common/constants';
+
+export function getOAuthTokenPackageParams(cookieHeaders: string | string[] | undefined) {
+  // In the future the token package will be stored in the login session. For now it's in a cookie.
+
+  // No cookie headers? No tokens
+  if (!cookieHeaders) {
+    return {};
+  }
+
+  // Take any cookie headers and split the individual cookies out, e.g. "_my_cookie=chocolateChip"
+  const cookiePayloads = [cookieHeaders].flat().flatMap((rawHeader) => rawHeader.split('; '));
```

Comment:
The combination of `flat()` and `flatMap()` was initially extremely confusing to me. If I'm understanding correctly, you're doing it here because of the possibility of `cookieHeaders` being either a string or array of string. Just IMO, I strongly think you should write that out explicitly instead of repeating flat to handle it for you:

```suggestion
  const cookieArray = Array.isArray(cookieHeaders) ? cookieHeaders : [cookieHeaders];
  const cookiePayloads = cookieArray.flatMap((rawHeader) => rawHeader.split('; '));
```

Reply:
Sure, I can split that out.